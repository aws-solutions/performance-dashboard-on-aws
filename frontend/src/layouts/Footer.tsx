/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import packagejson from "../../package.json";
import { useSettings } from "../hooks";
import { useTranslation } from "react-i18next";

function Footer() {
    const { settings, loadingSettings } = useSettings();
    const { t } = useTranslation();

    return (
        <footer>
            <div className="grid-container margin-bottom-9 text-base font-sans-sm">
                {loadingSettings ? (
                    ""
                ) : (
                    <>
                        <hr className="margin-top-9 border-base-lightest" />
                        {t("AdminFooter.HelpMessage")}{" "}
                        <a
                            href={`mailto:${
                                settings.adminContactEmailAddress ?? settings.contactEmailAddress
                            }?subject=${t("AdminFooter.PerformanceDashboardAssistance")}`}
                            className="usa-link"
                        >
                            {t("AdminFooter.ContactSupport")}
                        </a>
                        <span className="float-right">v{packagejson.version}</span>
                    </>
                )}
            </div>
        </footer>
    );
}

export default Footer;
