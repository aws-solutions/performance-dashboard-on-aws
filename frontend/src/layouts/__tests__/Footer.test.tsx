/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Footer from "../Footer";

describe("FooterLayout", () => {
    test("renders the component", async () => {
        const wrapper = render(<Footer />, { wrapper: MemoryRouter });
        expect(wrapper.container).toMatchSnapshot();
    });
});
