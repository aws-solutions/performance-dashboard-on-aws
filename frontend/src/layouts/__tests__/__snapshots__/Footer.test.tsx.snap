// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterLayout renders the component 1`] = `
<div>
  <footer>
    <div
      class="grid-container margin-bottom-9 text-base font-sans-sm"
    >
      <hr
        class="margin-top-9 border-base-lightest"
      />
      Having technical issues with the system?
       
      <a
        class="usa-link"
        href="mailto:admin1@aol.com?subject=Performance Dashboard Assistance"
      >
        Contact support.
      </a>
      <span
        class="float-right"
      >
        v
        v1.0.0
      </span>
    </div>
  </footer>
</div>
`;
